### PR TITLE
feat: order choice lists

### PR DESCRIPTION
**Pull request by Davidmattei**
|Q              |A  |
|---------------|---|
|Bug fix?       |no|
|New feature?   |yes|	
|BC breaks?     |no|	
|Deprecations?  |no|	
|Fixed tickets  |no|	

New property in choice document named: **choice_sort**
Possible values: **label_alpha** or **value_alpha**

### CODE REVIEW

File: FormConfig/FieldChoicesConfig.php [hunk 1]
```diff
@@ -68,6 +70,11 @@ public function getMaxLevel(): int
         return $this->calculateMaxLevel($this->values);
     }
 
+    public function setSort($sort): void
```
Comment (theus77) [inline FormConfig/FieldChoicesConfig.php:73]: Optional string?

File: FormConfig/FieldChoicesConfig.php [hunk 2]
```diff
@@ -118,6 +125,27 @@ private function combineValuesAndLabels(array $values, array $labels, array $cho
         }
 
         $list = \array_combine($this->getTopLevel($labels), $this->getTopLevel($values));
-        return \is_array($list) ? $list : [];
```
Comment (theus77) [inline FormConfig/FieldChoicesConfig.php:121]: What about the other levels?

File: FormConfig/FieldChoicesConfig.php (hunk 1 above)
Comment (Davidmattei) [reply FormConfig/FieldChoicesConfig.php:74]: this function should not be called with a null, sort is by default null. (updated constructor)

File: FormConfig/FieldChoicesConfig.php (hunk 2 above)
Comment (Davidmattei) [reply FormConfig/FieldChoicesConfig.php:121]: This function is called foreach level, but there was still a bug. The top level placeholder was used in all child selects. Fix in 950f923 and tested with 3 level nested select.

File: FormConfig/FieldChoicesConfig.php [hunk 3]
```diff
@@ -68,6 +71,11 @@ public function getMaxLevel(): int
         return $this->calculateMaxLevel($this->values);
     }
 
+    public function setSort($sort): void
```
Comment (theus77) [inline FormConfig/FieldChoicesConfig.php:74]: Then "setSort(string $sort)"?

File: FormConfig/FieldChoicesConfig.php [hunk 4]
```diff
@@ -118,6 +126,34 @@ private function combineValuesAndLabels(array $values, array $labels, array $cho
         }
 
         $list = \array_combine($this->getTopLevel($labels), $this->getTopLevel($values));
-        return \is_array($list) ? $list : [];
+
+        return $this->sort(\is_array($list) ? $list : []);
+    }
+
+    private function sort(array $list): array
```
Comment (theus77) [inline FormConfig/FieldChoicesConfig.php:133]: hint about the array?

**Review by theus77 — APPROVED** [review]

**Pull request merged**